### PR TITLE
Scroll to column

### PR DESCRIPTION
**Pull request by James300**
Associated Issue: #3892

### Current Issues

* Scroll to does not seems to be working
* Tests are breaking for `getOutOfScopeLocations`

I also need to clean some things up.

### Summary of Changes

Coming soon...

### Test Plan

* Search sources still opens source correctly
* Search files still opens source correctly
* Sources tree still opens source correctly
* Searching by files opens to line and column

### CODE REVIEW

File: src/components/ProjectSearch/TextSearch.js [hunk 1]
```diff
@@ -38,7 +38,9 @@ export default class TextSearch extends Component {
   }
 
   selectMatchItem(matchItem) {
-    this.props.selectSource(matchItem.sourceId, { line: matchItem.line });
+    this.props.selectSource(matchItem.sourceId, {
+      location: { line: matchItem.line, column: matchItem.column }
```
Comment (jasonLaster) [inline src/components/ProjectSearch/TextSearch.js]: this is a bit nicer
```js
const {line, column} = matchItem
this.props.selectSource(matchItem.sourceId, { location: { line, column }} )
```

File: src/actions/pause.js [hunk 2]
```diff
@@ -105,7 +105,10 @@ export function paused(pauseInfo: Pause) {
     }
 
     dispatch(
-      selectSource(frame.location.sourceId, { line: frame.location.line })
+      selectSource(frame.location.sourceId, {
```
Comment (jasonLaster) [inline src/actions/pause.js]:
```js
const {line, column} = frame.location;
selectSource(frame.location.sourceId, {location: {line, column}})
```

**Comment by jasonLaster** [comment]
Hey @James300, i did a quick scan of the tests and saw a couple common issues:

*  `TypeError: lineInfo.wrapClass is undefined`
* `"TypeError: location is undefined"`

I would get one test like `browser_dbg-iframes` passing w/ `yarn mochi browser_dbg-iframes` and then do a full run of the tests: `yarn mochi`.

File: src/actions/pause.js [hunk 3]
```diff
@@ -235,9 +234,8 @@ export function selectFrame(frame: Frame) {
 
     dispatch({ type: "SELECT_FRAME", frame, scopes });
 
-    dispatch(
-      selectSource(frame.location.sourceId, { line: frame.location.line })
-    );
+    const { line, column } = frame.location;
+    dispatch(selectSource(frame.location.sourceId, { line, column }));
```
Comment (bomsy) [inline src/actions/pause.js]: and this also ..?

File: src/actions/pause.js [hunk 4]
```diff
@@ -104,9 +104,8 @@ export function paused(pauseInfo: Pause) {
       dispatch(evaluateExpressions());
     }
 
-    dispatch(
-      selectSource(frame.location.sourceId, { line: frame.location.line })
-    );
+    const { line, column } = frame.location;
+    dispatch(selectSource(frame.location.sourceId, { line, column }));
```
Comment (bomsy) [inline src/actions/pause.js]: shouldn't this be `dispatch(selectSource(frame.location.sourceId, { location: frame.location }));`

**Comment by jasonLaster** [comment]
Thanks @bomsy these look really good. I see some flow issues which i think will help us get to the bottom of the pretty-print mochitest and some of the others!

File: src/components/Editor/index.js [hunk 5]
```diff
@@ -201,24 +202,38 @@ class Editor extends PureComponent {
...
     // If the location is different and a new line is requested,
     // update the pending jump line. Note that if jumping to a line in
     // a source where the text hasn't been loaded yet, we will set the
     // line here but not jump until rendering the actual source.
-    if (prevProps.selectedLocation !== selectedLocation) {
-      if (selectedLocation && selectedLocation.line != undefined) {
+
+    if (
+      !prevProps.selectedLocation ||
+      prevProps.selectedLocation.location !== selectedLocation.location
```
Comment (jasonLaster) [inline src/components/Editor/index.js]: will this work? 

* do we need to use: https://lodash.com/docs/#isEqual or should identity comparisons be enough
* naive question: why is it `prevProps.selectedLocation.location` and not `prevProps.selectedLocation`? Are we messing something up?

File: src/components/Editor/index.js [hunk 6]
```diff
@@ -201,24 +202,38 @@ class Editor extends PureComponent {
...
     // Only update and jump around in real source texts. This will
     // keep the jump state around until the real source text is
     // loaded.
-    if (selectedSource && selectedSource.has("text")) {
-      this.flashLine();
+    if (
+      !prevProps.selectedLocation ||
+      !prevProps.selectedLocation.location ||
+      (prevProps.selectedLocation &&
+        prevProps.selectedLocation.location &&
```
Comment (jasonLaster) [inline src/components/Editor/index.js]: why do we now care about `prevProps.selectedLocation.location`?

Comment (bomsy) [reply src/components/Editor/index.js]: good point!

File: src/reducers/sources.js [hunk 7]
```diff
@@ -71,15 +71,15 @@ function update(
...
+        ...action.location,
         url: action.source.url
       };
 
       prefs.pendingSelectedLocation = location;
       return state
         .set("selectedLocation", {
           sourceId: action.source.id,
-          line: action.line
+          location: action.location
```
Comment (jasonLaster) [inline src/reducers/sources.js]: i think this is the source of the redundancy... 
we should probably just be doing

```js
.set("selectedLocation", action.location)
```

Comment (bomsy) [reply src/reducers/sources.js]: I'm thinking
```js 
.set("selectedLocation", {
   sourceId: action.source.id,
   ...action.location
})
```
as we need the `sourceId` property

**Pull request merged**